job/common/src/main/java/alluxio/job/JobConfig.java

```diff
@@ -14,16 +14,31 @@
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonTypeInfo;
 
+import javax.annotation.Nullable;
 import java.io.Serializable;
 
 /**
  * A job configuration. All the subclasses are both Java and JSON serializable.
  */
 @JsonIgnoreProperties(ignoreUnknown = true)
 @JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "@type")
-public interface JobConfig extends Serializable {
+public abstract class JobConfig implements Serializable {
```

Comment:
I would prefer this to be an interface if possible. I'm not convinced we need to limit the size of `getDescription` here, unless the bottleneck we are trying to avoid is too large of payload?

Reply:
I took a look at when @gpang was debugging when he decided to print out this and it was in the 100,000s of characters but I can limit from the client instead.

Reply:
Yeah if data transfer / serialization is not the concern and the concern is readability we should limit on the client side.